src/Gridify/GridifyMapper.cs

```diff
@@ -220,6 +220,26 @@ public IEnumerable<IGMap<T>> GetCurrentMaps()
    {
       return _mappings;
    }
+   
+   public IEnumerable<IGMap<T>> GetCurrentMapsByType(HashSet<Type> targetTypes)
+   {
+      foreach (var map in _mappings)
+      {
+         if (map.To.Body is UnaryExpression unaryExpression)
+         {
+            if (targetTypes.Contains(unaryExpression.Operand.Type))
+            {
+               yield return map;
+            }
+         }
+         // TODO: need to implement other types that aren't unary expression
```

Comment:
This needs to be completed to support and test all types; currently, it only works with unary expressions. Adding a couple of additional unit tests for known dotnet types and a few custom types is also required.

Reply:
I ppdated the method a bit further, I am not sure what other types of expressions are out there. Where can I find the list? 

Probably my lack of depth in understanding expressions. This approach isn't looking so hot anymore, where we have to figure out each expression type to find the property type.

Reply:
Handling all types of expressions isn't necessary; I believe these three cover at least 80% of the types, and so far, it looks very promising. Perhaps we should consider adding another test case to determine if we can support custom types. Lastly, the PR seems fine, but the documentation needs to be updated.